ee/clickhouse/queries/actor_base_query.py

```diff
@@ -0,0 +1,114 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Tuple,
+    TypedDict,
+    Union,
+)
+
+from django.db.models.query import QuerySet
+
+from ee.clickhouse.client import sync_execute
+from posthog.models import Entity, Filter, Team
+from posthog.models.group import Group
+from posthog.models.person import Person
+
+
+class SerializedPerson(TypedDict):
+    type: Literal["person"]
+    id: str
+    created_at: Optional[str]
+    properties: Dict[str, Any]
+    is_identified: Optional[bool]
+    name: str
+    distinct_ids: List[str]
+
+
+class SerializedGroup(TypedDict):
+    type: Literal["group"]
+    group_key: str
+    created_at: Optional[str]
+    properties: Dict[str, Any]
+
+
+SerializedActor = Union[SerializedPerson, SerializedGroup]
+Actor = Union[Person, Group]
+
+
+class ActorBaseQuery:
+    aggregating_by_groups = False
+
+    def __init__(self, team: Team, filter: Filter, entity: Optional[Entity] = None):
+        self.team = team
+        self.entity = entity
+        self.filter = filter
+
+        if self.entity and self.entity.math == "unique_group":
+            self.aggregating_by_groups = True
+
+    def groups_query(self) -> Tuple[str, Dict]:
```

Comment:
Q: Why do subclasses need to implement both? Existing queries are already using `aggregation_target` as the column to return from event_query, which would be the group key in one case and person_id in the other.

Reply:
That said, looking at the only implementation these are really similar. I'll need to read through the code with a fresh head tomorrow, I _think_ there might be ways to simplify this further.

Reply:
Yup you're right. I combined the functions